scalafmt-core/shared/src/main/scala/org/scalafmt/internal/FormatOps.scala

```diff
@@ -1080,13 +1081,19 @@ class FormatOps(val tree: Tree, val initStyle: ScalafmtConfig) {
 
   def getSpaceAndNewlineAfterCurlyLambda(
       newlines: Int
-  )(implicit style: ScalafmtConfig): (Boolean, NewlineT) =
+  )(implicit style: ScalafmtConfig): (Boolean, NewlineT) = {
+    val tryToSquashLambdaInOneLine =
+      style.newlines.sourceIs(Newlines.fold) || newlines == 0
```

Comment:
the problem here is that you are looking at `newlines` when source is `unfold`, and we can't do that. `sourceIgnored` is better.